src/www/ui/page/AdminLicenseCandidate.php

```diff
@@ -262,6 +279,32 @@ private function mergeCandidate($candidate, $suggest, $vars)
     return true;
   }
 
+  protected function doDeleteCandidate($rfPk)
+  {
+    $dbManager = $this->getObject('db.manager');
+    $stmt = __METHOD__.".$rfPk";
+    $dbManager->prepare($stmt, "SELECT DISTINCT(uploadtree_fk) FROM clearing_event "
+                               ."WHERE rf_fk=$1 AND removed=false AND uploadtree_fk NOT IN "
+                               ."(SELECT DISTINCT(uploadtree_fk) FROM clearing_event WHERE "
+                               ."rf_fk=$1 AND removed=TRUE);");
+    $result = $dbManager->execute($stmt, array($rfPk));
+    $dataFetch = $dbManager->fetchAll($result);
+    $dbManager->freeResult($result);
+    if(empty($dataFetch)){
+      $dbManager->getSingleRow('DELETE FROM license_candidate WHERE rf_pk=$1', array($rfPk), __METHOD__.".delete");
+      return new Response('true', Response::HTTP_OK, array('Content-type'=>'text/plain'));
+    }else{
+      $treeDao = $this->getObject('dao.tree');
+      $Path  = "Cannot remove license, because it is in use at \n\n";
+      foreach($dataFetch as $cnt => $uploadTreeFk){
+        $Path .= $cnt+1;
+        $Path .= ") ".$treeDao->getFullPath($uploadTreeFk['uploadtree_fk'], 'uploadtree')."\n";
```

Comment:
please use a real html `<ol>`-list instead of `1) .... 2)....`.

Reply:
Javascript alert box will not render any kind of html tags. So plan text is used here

Reply:
Below I suggested to use the jquery-ui alert and confirm boxes, which support html tags.